numpy/lib/function_base.py

```diff
@@ -1917,6 +1917,128 @@ def diff(a, n=1, axis=-1):
         return da
 
 
+def ratio(a, n=1, axis=-1, true=True, zero_div=None):
+    """
+    Calculate the n-th geometric difference along the given axis.
+
+    The first ratio is given by ``out[n] = a[n+1] // a[n]`` along
+    the given axis, higher differences are calculated by using `ratio`
+    recursively.
+
+    Parameters
+    ----------
+    a : array_like
+        Input array
+    n : int, optional
+        The number of times values are differenced. If zero, the input
+        is returned as-is.
+    axis : int, optional
+        The axis along which the difference is taken, default is the
+        last axis.
+    true : bool, optional
+        Whether or not to use true division or to preserve the type.
+        If True, the ratio will use `/` and the result will be an
+        inexact type. If False, the type will be preserved with the
+        `//` operator. Default is True.
+    zero_div : scalar, optional
+        A scalar to replace divisions by zero with. If omitted, the
+        default value will be used in the result and a warning will be
+        issued as usual.
```

Comment:
Because sometimes you will want to put in a finite flag value instead of getting a warning and `inf`. In fact, you can even set this to `np.inf` to get the same result but with no warning.